File: c++/data_structures/stack.cpp
#include <vector>

using std::vector;

template <class T>
class Stack {
    // push, pop, peek, size
    void push(T item) {
        items.push_back(item);
    }

    T pop () {
        return items.pop_back();
    }

    size_t size () {
        return items.size();
    }

    T peek () {
        return items[-1];
    }

    private:
        vector<T> items;
};

int main (int argc, char *argv[]) {
    Stack<int> s;
    return 0;
}

File: js/data_structures/stack.ts
class Stack<T> {
    #items: T[]
    constructor() {
        this.#items = [];
    }

    //push, pop, peek, size
    push(item: T) {
        this.#items.push(item);
    }

    pop() {
        return this.#items.pop();
    }

    size() {
        return this.#items.length;
    }

    peek() {
        return this.#items[this.#items.length-1]
    }
}

const t_stack = new Stack();
